monai/data/meta_obj.py

```diff
@@ -103,71 +103,30 @@ def flatten_meta_objs(*args: Iterable):
-        """
-        Copy attributes from the first in a list of `MetaObj`. In the case of
-        `torch.add(a, b)`, both `a` and `b` could be `MetaObj` or something else, so
-        check them all. Copy the first to `self`.
-
-        We also perform a deep copy of the data if desired.
-
-        Args:
-            attributes: a sequence of strings corresponding to attributes to be copied (e.g., `['meta']`).
-            input_objs: an iterable of `MetaObj` instances. We'll copy the attribute from the first one
-                that contains that particular attribute.
-            defaults: If none of `input_objs` have the attribute that we're
-                interested in, then use this default value/function (e.g., `lambda: {}`.)
-                the defaults must be the same length as `attributes`.
-            deep_copy: whether to deep copy the corresponding attribute.
-
-        Returns:
-            Returns `None`, but `self` should be updated to have the copied attribute.
-        """
-        found = [False] * len(attributes)
-        for i, (idx, a) in itertools.product(input_objs, enumerate(attributes)):
-            if not found[idx] and hasattr(i, a):
-                setattr(self, a, MetaObj.copy_items(getattr(i, a)) if deep_copy else getattr(i, a))
-                found[idx] = True
-            if all(found):
-                return
-        for a, f, d in zip(attributes, found, defaults):
-            if not f:
-                setattr(self, a, d() if callable(defaults) else d)
-        return
-
     @staticmethod
     def copy_items(data):
-        """deepcopying items"""
-        if isinstance(data, (int, float, str, bool, Enum)):
-            return data
-        if isinstance(data, np.ndarray):
-            return np.array(data, copy=True)
+        """returns a copy of the data. list and dict are shallow copied for efficiency purposes."""
+        if isinstance(data, (list, dict, np.ndarray)):
+            return data.copy()
         if isinstance(data, torch.Tensor):
             return data.detach().clone()
-        if isinstance(data, (list, tuple)):
-            return type(data)(MetaObj.copy_items(x) for x in data)
-        if isinstance(data, dict):
-            return {k: MetaObj.copy_items(v) for k, v in data.items()}
         return deepcopy(data)
 
-    def _copy_meta(self, input_objs, deep_copy=False) -> None:
+    def copy_meta_from(self, input_objs, copy_attr=True) -> None:
         """
-        Copy metadata from an iterable of `MetaObj` instances. For a given attribute, we copy the
-        adjunct data from the first element in the list containing that attribute.
+        Copy metadata from a `MetaObj` or an iterable of `MetaObj` instances.
 
         Args:
             input_objs: list of `MetaObj` to copy data from.
-
+            copy_attr: whether to copy each attribute with `MetaObj.copy_item`.
+                note that if the attribute is a nested list or dict, only a shallow copy will be done.
```

Comment:
Seems even if `copy_attr = True`, it still shallow copy the numpy array or tensor?
So if I really want to deepcopy, can I use `a = MetaTensor(...), MetaTensor b = deepcopy(a)` or `b.meta = deepcopy(a.meta)`?

Thanks.

Reply:
that's a good point, now it's updated to deepcopy in two cases when a is a MetaTensor, `b = MetaTensor(a)` and `b = a.clone()`, what do you think?

Reply:
OK, the solution looks good to me. Please document it clearly, I feel it's easy to be confused..

Thanks.